src/openrct2/object/ImageTable.cpp

```diff
@@ -30,6 +30,8 @@
 using namespace OpenRCT2;
 using namespace OpenRCT2::Drawing;
 
+static thread_local std::map<u8string, std::unique_ptr<Object>> _objDataCache = {};
```

Comment:
I'm not sure about this being `thread_local`. Don't you want this to be shared, since it's essentially loaded from disk? If multiple threads use this (which I don't know is the case), it'll load the object file for each thread. I/O should be done as little as possible.

Reply:
No, you don’t want this to be shared - it causes the game to crash in the destructors of the objects. AFAIK, loading objects _is_ multithreaded, but on the level of the object itself, and not the individual images of it.

Reply:
@Gymnasiast you do want it to be shared (unless performance drops), it just needs to be thread safe to prevent it from crashing. We use mutex in other places to lock areas of code that should not run concurrently.

Reply:
Would it make any difference here, though? The cache is emptied every time a JSON object has all its images loaded, so another object that happens to reference the same .DAT for its images will not be able to take advantage of it anyway. Actually, I could dispense with the static variable and just make it a local variable passed via arguments. That might actually be a little bit cleaner.

It would of course be different if the cache was used for all loaded objects, although that would only be advantageous if many .DAT files are referenced by multiple JSON objects, which I don’t think is the case. And it would take a lot of additional memory, so that might make it slower.

Reply:
I see, okay then I suppose it isn't so beneficial then.
I need to sit down and look at the changes more closely anyway and compare them with the work I did for graphics asset packs.

Reply:
That’s all right, but if that work doesn’t make it for v0.4.2, I’d like to have this one merged. It solves a potentially massive performance issue for people without RCT1 linked, and the commit is easily reversible.